runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -213,138 +213,21 @@ public static String jobToString(Job job) {
   }
 
   /**
-   * A {@link TransformTranslator} knows how to translate
-   * a particular subclass of {@link PTransform} for the
-   * Cloud Dataflow service. It does so by
-   * mutating the {@link TranslationContext}.
+   * A {@link TransformTranslator} knows how to translate a particular subclass of {@link
+   * PTransform} for the Cloud Dataflow service. It does so by mutating the {@link Translator}.
    */
   public interface TransformTranslator<TransformT extends PTransform> {
-    void translate(TransformT transform,
-                          TranslationContext context);
+    void translate(TransformT transform, Translator context);
   }
 
-  /**
-   * The interface provided to registered callbacks for interacting
-   * with the {@link DataflowRunner}, including reading and writing the
-   * values of {@link PCollection}s and side inputs ({@link PCollectionView}s).
```

Comment:
Since many of the `TransformTranslator`s are within this class that means that they have access to the private methods of the `Translator`. By using the interface `TranslationContext`, we controlled which methods they had access to even inside this class.

If you want to continue down this path, please save the javadocs from this interface and move them to the Translator class and make the Translator class a package private class in another file so the private visibility rules still apply.

If you want to keep the `TranslationContext`, you could do the same thing your doing right now by refactoring out the methods that should be on a `StepContext` and returning a `StepContext` from `addStep`.